src/plugins/Wallet/wallet.ts

```diff
@@ -18,21 +19,20 @@ export function getWalletProvider() {
     return walletAPI
 }
 const memoGetWalletBalance = memoizePromise(
-    (addr: string) => {
-        return getWalletProvider()
-            .queryBalance(addr)
-            .then((x) => onWalletBalanceUpdated(addr, x))
+    async (addr: string) => {
+        const x = await getWalletProvider().queryBalance(addr)
+        return onWalletBalanceUpdated(addr, x)
     },
     (x) => x,
 )
 const memoQueryERC20Token = memoizePromise(
-    (addr: string, erc20Addr: string) => {
-        return getWalletProvider()
-            .queryERC20TokenBalance(addr, erc20Addr)
-            .then((x) => onWalletERC20TokenBalanceUpdated(addr, erc20Addr, x))
-            .catch(() => {
-                // do nothing
-            })
+    async (addr: string, erc20Addr: string) => {
+        try {
+            const x = await getWalletProvider().queryERC20TokenBalance(addr, erc20Addr)
+            return onWalletERC20TokenBalanceUpdated(addr, erc20Addr, x)
```

Comment:
If you use `return`, error in `onWalletERC20TokenBalanceUpdated` won't be caught by the catch clause.

Reply:
ACK

Reply:
It seems actually the same according to my JS knowledge...

Reply:
`return await expr` is ok

Reply:
Ah, so it is.

Reply:
@SunriseFox try the following code:
```js
async function a() {
    try { return Promise.reject() }
    catch { return 'ok' }
}
await a() // uncaught promise rejection
async function b() {
    try { return await Promise.reject() }
    catch { return 'ok' }
}
await b() //"ok"
```